nsqd/protocol_v2.go

```diff
@@ -92,6 +93,7 @@ func (p *protocolV2) IOLoop(conn net.Conn) error {
 
 			sendErr := p.Send(client, frameTypeError, []byte(err.Error()))
 			if sendErr != nil {
+				p.ctx.nsqd.logf("ERROR: [%s] - %s%s", client, sendErr, ctx)
```

Comment:
Let me clarify without editing - if `Exec` returned a `FatalClientError` we probably want to returned error to be the one which determined `IOLoop` is going to exit. In this case, `sendErr` is secondary.

If `Exec` returned another type of error it may not have necessarily exited, but we are now that `p.Send` has failed. In this case, if we want to return the error which caused the loop to exit it would be `sendErr`. A few if statements and we could have this behavior.

I think, please correct me if I'm wrong, but at this time all errors returned by `Exec` are `FatalClientErr`.